[Feat] Badge 컴포넌트 생성

aken-you opened the pull request:
# 관련 이슈 번호

close #39

# 설명

variant="num"일 경우, number prop을 필수값으로 설정하기 위해 에러 처리했습니다.
또한 variant="text"일 경우에도 text prop 입력을 하도록 에러 처리하였습니다.

aken-you committed:
feat[#39] : Badge 컴포넌트 생성

aken-you committed:
feat[#39] : Badge 스토리북 작성

yonghyeun commented:
### ✅ Test-Storybook Results - Success
```
PASS browser: chromium ../../../../../tmp/2859418a783f3c25c8bd55c9cdf1c2c0/shared-button.test.js
PASS browser: chromium ../../../../../tmp/2859418a783f3c25c8bd55c9cdf1c2c0/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/2859418a783f3c25c8bd55c9cdf1c2c0/shared-input.test.js

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        5.921 s
Ran all test suites.

```
🚀 [스토리북 확인하기](https://66bae48876a5167719ee1778-jqrdxsprac.chromatic.com)

yonghyeun commented on src/shared/ui/badge/Badge.tsx:
```diff
@@ -0,0 +1,41 @@
+
+  const textStyles = "text-grey-0 body-3";
+
+  if (variant === "num") {
```
&& 연산자로 깔끔하게 해보면 어떨까요  👁️

yonghyeun commented on src/shared/ui/badge/Badge.tsx:
```diff
@@ -0,0 +1,41 @@
+    );
+  }
+
+  if (text === undefined)
```
varient === num 이고 text 가 undefined 일 경우에 문제가 생길 거 같습니다 👁️👁️

yonghyeun reviewed (commented):
덤벙거리는 제가 일으킬 수 있는 휴먼 에러들이 많아서 고생이 많으십니다 👁️👁️

aken-you committed:
refactor[#39] : Badge의 prop에서 children 추가

- 기존에 text, number prop을 삭제
- children의 타입을 string으로 좁힘
- 만약 숫자로만 구성된 text라면 baseStyles.num을,  아니라면 baseStyles.text를 적용

yonghyeun commented:
### ✅ Test-Storybook Results - Success
```
PASS browser: chromium ../../../../../tmp/0fbf0933e88331cfdc1eba229ad19991/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/0fbf0933e88331cfdc1eba229ad19991/shared-button.test.js
PASS browser: chromium ../../../../../tmp/0fbf0933e88331cfdc1eba229ad19991/shared-input.test.js

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        4.17 s
Ran all test suites.

```
🚀 [스토리북 확인하기](https://66bae48876a5167719ee1778-jqrdxsprac.chromatic.com)

aken-you commented:
구조를 다시 바꿨어요!

Badge 안에 들어갈 내용을 prop으로 받게 되면, 컴포넌트 내부에서 많은 분기처리가 필요하겠다 싶었어요.
Badge 기능에 변경이 일어난다면, 많은 분기처리 속에서 어떤 부분을 수정해야하는지 바로 찾을 수 있을까? 고민했습니다.

그래서 props에서 text, number을 제거하고, children으로 대체했습니다.
대신 children 타입을 string으로 좁혀서 종류를 다음과 같이 구분했습니다.

1. children이 없는 경우 => dot
2. children이 숫자로만 구성되어 있는 경우 => num
3. children이 숫자로만 구성되어 있지 않은 경우 => text

yonghyeun commented on src/shared/ui/badge/Badge.tsx:
```diff
@@ -0,0 +1,32 @@
+
+  const textStyles = "text-grey-0 body-3";
+
+  const hasChildren = children !== undefined;
```
상단에서 isDot 의 얼리리턴 조건문을 통해 children이 없음은 명확해지기 때문에 hasChildren은 없어도 되지 않을까요 ? 👁️👁️

yonghyeun commented on src/shared/ui/badge/Badge.tsx:
```diff
@@ -0,0 +1,32 @@
+  const textStyles = "text-grey-0 body-3";
+
+  const hasChildren = children !== undefined;
+  const isNum = hasChildren && /^[0-9]+$/.test(children);
```
들어올 수 있는 children 타입을 string 으로 좁히고 정규표현식으로 걸러내신건 정말 생각도 못한 엄청난 방법이네요 !! 

완전 감탄스럽습니다 🔥🫠

yonghyeun reviewed (commented):
무릎을 탁 치는 코드입니다

스트링으로 좁히고 정규표현식은 정말 생각도 못한 ..🫠

aken-you committed:
refactor[#39] : 필요 없는 변수 삭제

- 위에서 isDot으로 early return 하기 때문에 hasChildren 삭제

yonghyeun commented:
### ✅ Test-Storybook Results - Success
```
PASS browser: chromium ../../../../../tmp/3ace9cb0ce7f878016658aaff4709b34/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/3ace9cb0ce7f878016658aaff4709b34/shared-input.test.js
PASS browser: chromium ../../../../../tmp/3ace9cb0ce7f878016658aaff4709b34/shared-button.test.js

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        4.415 s
Ran all test suites.

```
🚀 [스토리북 확인하기](https://66bae48876a5167719ee1778-jqrdxsprac.chromatic.com)
